src/SerialLibrary/__init__.py

```diff
@@ -524,12 +525,50 @@ def read_until(self, terminator=LF, size=None, encoding=None, port_locator=None)
         """
         if size is not None:
             size = float(size)
-        if terminator != LF:
+        if terminator != LF and not isinstance(terminator, (bytes, bytearray)):
             terminator = self._encode(terminator)
         return self._decode(
             self._port(port_locator).read_until(terminator=terminator, size=size),
             encoding)
 
+    def read_until_pattern(self, pattern_regexp, terminator=LF, size=None, encoding=None, port_locator=None):
+        """
+        Read until a pattern regex definition will be found, size exceeded or timeout.
+
+        If `encoding` is not given, default encoding is used.
+        Note that encoding affects terminator too, so if you want to use
+        character 'X' as terminator and encoding=hexlify (default), you should
+        call this keyword as Read Until terminator=58.
+        """
+        if size is not None:
+            size = float(size)
+        if terminator != LF and not isinstance(terminator, (bytes, bytearray)):
+            terminator = self._encode(terminator)
+        bytes_read = 0
+        regexp = re.compile(pattern_regexp, re.MULTILINE)
+        old_timeout = self._port(port_locator).timeout
+        start_time = time.time()
+        self._port(port_locator).timeout = 0.1
```

Comment:
Why this value of `0.1`? Please clarify with an inline comment if you have a good reason behind it.